tools/include/eosio/abigen.hpp

```diff
@@ -530,5 +626,143 @@ namespace eosio { namespace cdt {
          std::set<abi_table>                   ctables;
          std::map<std::string, std::string>    rcs;
          std::set<const clang::Type*>          evaluated;
+
+         bool is_kv_table(const clang::CXXRecordDecl* decl) {
```

Comment:
This function and `get_value_from_decltype` and `internal_types` are generic enough that they should live in `gen.hpp`.